themes/bootstrap3/less/bootstrap.less

```diff
@@ -58,6 +58,9 @@ label { cursor: pointer; }
 .icon-bar { background-color: #888; }
 
 // Icons vs. Underline fix
+.btn {
+  text-decoration: none;
+}
 .btn-link {
   &:extend(.icon-link);
```

Comment:
As long as btn-link still has the underline, this is exactly the change I was hoping for!